[ONNX] Add linspace symbolic (#58854)

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #60249 [ONNX] Update documentation (#58712)
* #60248 [ONNX] shape type inference fixes for control flow (#59319)
* #60247 [ONNX] Extend chunk for dynamic chunk values (#59644)
* **#60246 [ONNX] Add linspace symbolic (#58854)**
* #60245 [ONNX] Fix sum export with attribute keepdims (#59316)
* #60244 [ONNX] Fix shape inference for large model (#59320)
* #60243 [ONNX] Handle onnx::Size in ComputeConstant folding (#59122)

* Adds support for linspace op
* Modifies arange symbolic in opset 9 to replicate the same behavior in which dtype is determined (similar to opset 11) as in https://pytorch.org/docs/stable/generated/torch.arange.html
* Enabled some arange unit tests which were disabled for opset 9

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

Differential Revision: [D29494911](https://our.internmc.facebook.com/intern/diff/D29494911)

shubhambhokare1 committed:
[ONNX] Add linspace symbolic (#58854)

* Adds support for linspace op
* Modifies arange symbolic in opset 9 to replicate the same behavior in which dtype is determined (similar to opset 11) as in https://pytorch.org/docs/stable/generated/torch.arange.html
* Enabled some arange unit tests which were disabled for opset 9

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add linspace symbolic (#58854)"


* Adds support for linspace op
* Modifies arange symbolic in opset 9 to replicate the same behavior in which dtype is determined (similar to opset 11) as in https://pytorch.org/docs/stable/generated/torch.arange.html
* Enabled some arange unit tests which were disabled for opset 9

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add linspace symbolic (#58854)"


* Adds support for linspace op
* Modifies arange symbolic in opset 9 to replicate the same behavior in which dtype is determined (similar to opset 11) as in https://pytorch.org/docs/stable/generated/torch.arange.html
* Enabled some arange unit tests which were disabled for opset 9

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add linspace symbolic (#58854)"


* Adds support for linspace op
* Modifies arange symbolic in opset 9 to replicate the same behavior in which dtype is determined (similar to opset 11) as in https://pytorch.org/docs/stable/generated/torch.arange.html
* Enabled some arange unit tests which were disabled for opset 9

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add linspace symbolic (#58854)"


* Adds support for linspace op
* Modifies arange symbolic in opset 9 to replicate the same behavior in which dtype is determined (similar to opset 11) as in https://pytorch.org/docs/stable/generated/torch.arange.html
* Enabled some arange unit tests which were disabled for opset 9

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add linspace symbolic (#58854)"


* Adds support for linspace op
* Modifies arange symbolic in opset 9 to replicate the same behavior in which dtype is determined (similar to opset 11) as in https://pytorch.org/docs/stable/generated/torch.arange.html
* Enabled some arange unit tests which were disabled for opset 9

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add linspace symbolic (#58854)"


* Adds support for linspace op
* Modifies arange symbolic in opset 9 to replicate the same behavior in which dtype is determined (similar to opset 11) as in https://pytorch.org/docs/stable/generated/torch.arange.html
* Enabled some arange unit tests which were disabled for opset 9

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add linspace symbolic (#58854)"


* Adds support for linspace op
* Modifies arange symbolic in opset 9 to replicate the same behavior in which dtype is determined (similar to opset 11) as in https://pytorch.org/docs/stable/generated/torch.arange.html
* Enabled some arange unit tests which were disabled for opset 9

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

[ghstack-poisoned]

SplitInfinity commented:
@splitinfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29494911).

BowenBao committed:
Update on "[ONNX] Add linspace symbolic (#58854)"


* Adds support for linspace op
* Modifies arange symbolic in opset 9 to replicate the same behavior in which dtype is determined (similar to opset 11) as in https://pytorch.org/docs/stable/generated/torch.arange.html
* Enabled some arange unit tests which were disabled for opset 9

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

Differential Revision: [D29494911](https://our.internmc.facebook.com/intern/diff/D29494911)

[ghstack-poisoned]

SplitInfinity commented:
@splitinfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29494911).
